src/battle_util.c

```diff
@@ -10217,3 +10217,47 @@ bool32 CanTargetBattler(u8 battlerAtk, u8 battlerDef, u16 move)
         return FALSE;   // Pokémon affected by Heal Block cannot target allies with Pollen Puff
     return TRUE;
 }
+
+bool32 ShouldRestoreItem(u16 itemId)
+{
+    u16 holdEffect = ItemId_GetHoldEffect(itemId);
+
+    switch (holdEffect)
+    { // Unsure about Absorb Bulb-type items, terrain seeds, Weakness/Blunder Policy, and Throat Spray.
+        case HOLD_EFFECT_RESTORE_HP:
+        case HOLD_EFFECT_CURE_PAR:
+        case HOLD_EFFECT_CURE_SLP:
+        case HOLD_EFFECT_CURE_PSN:
+        case HOLD_EFFECT_CURE_BRN:
+        case HOLD_EFFECT_CURE_FRZ:
+        case HOLD_EFFECT_RESTORE_PP:
+        case HOLD_EFFECT_CURE_CONFUSION:
+        case HOLD_EFFECT_CURE_STATUS:
+        case HOLD_EFFECT_CONFUSE_SPICY:
+        case HOLD_EFFECT_CONFUSE_DRY:
+        case HOLD_EFFECT_CONFUSE_SWEET:
+        case HOLD_EFFECT_CONFUSE_BITTER:
+        case HOLD_EFFECT_CONFUSE_SOUR:
+        case HOLD_EFFECT_ATTACK_UP:
+        case HOLD_EFFECT_DEFENSE_UP:
+        case HOLD_EFFECT_SPEED_UP:
+        case HOLD_EFFECT_SP_ATTACK_UP:
+        case HOLD_EFFECT_SP_DEFENSE_UP:
+        case HOLD_EFFECT_CRITICAL_UP:
+        case HOLD_EFFECT_RANDOM_STAT_UP:
+        case HOLD_EFFECT_EVASION_UP:
+        case HOLD_EFFECT_RESTORE_STATS:
+        case HOLD_EFFECT_MENTAL_HERB:
+        case HOLD_EFFECT_POWER_HERB:
+        case HOLD_EFFECT_RESIST_BERRY:
+        case HOLD_EFFECT_RESTORE_PCT_HP:
+        case HOLD_EFFECT_MICLE_BERRY:
+        case HOLD_EFFECT_CUSTAP_BERRY:
+        case HOLD_EFFECT_ROWAP_BERRY:
+        case HOLD_EFFECT_KEE_BERRY:
+        case HOLD_EFFECT_MARANGA_BERRY:
+        case HOLD_EFFECT_GEMS:
+            return TRUE;
+    }
+    return FALSE;
```

Comment:
Shouldn't this be backwards?

Reply:
Should it?

The function is called `ShouldRestoreItem`. If the held item of the item that is passed through it is any of the ones listed in the `switch`, the function would return `TRUE` to signal that the item should be restored. Otherwise it returns `FALSE` to signal the opposite.

Going by what's in `TryRestoreStolenItems`, that seems to make sense? :eyes:

Reply:
What I mean is that it looks like those item effects should be the ones that *don't* get restored, eg. Gems.

Reply:
Ah, I think you're right. `TryRestoreStolenItems` is called in relation to `B_TRAINERS_KNOCK_OFF_ITEMS == TRUE` which signals that an item that isn't a berry *(I feel it's more accurate to say "an item that is not consumed on use")* should be restored.

All the hold effects listed in `ShouldRestoreItem` seem to belong to items that are consumed on use, so if the held effect of an item that is passed through that function matches any of those, it should return `FALSE` :eyes: